zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -204,6 +266,50 @@ object Http {
     dd: Http[R, EE, A, BB],
   )                                                                             extends Http[R, EE, A, BB]
 
+  private final case class Provide[R, E, A, B](self: Http[R, E, A, B], r: R) extends Http[Any, E, A, B] {
+    override private[zhttp] def execute(a: A): HttpResult[Any, E, B] =
```

Comment:
`execute` was supposed to be final (miss from my side).
You will loose out on stacksafety if you override it.

Reply:
Gotcha.

The only reason why I use it is to get around an (possibly incorrect) exhaustivity warning on Scala3, see d45475f. Would be happy to just inline all impls if someone can point me to another way around this!

Reply:
1a119f3 is the only way around this I've found that also makes the compiler happy. Can't say it's super pretty (and not sure if extracting `doProvide` keeps it stack-safe?). If you have any other ideas of how to address this I'd be very grateful 🙏

Reply:
Oh. And that doesn't seem to work for 2.13. :/

Reply:
Changing all of them like so:

```scala
+      case p @ ProvideLayer(_, _)     => p.doProvide
+      // case p: ProvideLayer[_, _, _, _, _, _] => p.doProvide
```

works for both versions but emits warnings in Scala 3. 🤷